baseapp/baseapp_payments/views.py

```diff
@@ -111,3 +124,34 @@ def retrieve(self, request, remote_customer_id=None):
         if not customer:
             raise NotFound("Customer not found")
         return Response(customer, status=200)
+
+
+class StripePaymentMethodViewset(viewsets.ViewSet):
+    permission_classes = [IsAuthenticated]
+    serializer_class = StripePaymentMethodSerializer
+
+    def list(self, request):
+        remote_customer_id = request.query_params.get("customer_id")
+        if not remote_customer_id:
+            raise ValidationError("Missing customer_id")
+
+        try:
+            payment_methods = StripeService().get_customer_payment_methods(remote_customer_id)
+            serializer = self.serializer_class(payment_methods, many=True)
+            return Response(serializer.data, status=200)
+        except Exception as e:
+            logger.exception("Failed to retrieve payment methods: %s", e)
+            return Response({"error": "An internal error has occurred"}, status=500)
+
+    def create(self, request):
+        serializer = self.serializer_class(data=request.data)
```

Comment:
Also here, may have missed it again, but where are we validating that the customer_id belongs to the authenticated user?